### PR TITLE
Alec k/appeals 26730

### PR DESCRIPTION
**Pull request by AKeyframe**
Resolves [APPEALS-26730](https://jira.devops.va.gov/browse/APPEALS-26730) Update Rails to 6.0.6.1

### Acceptance Criteria
See documentation produced in [APPEALS-26727](https://jira.devops.va.gov/browse/APPEALS-26727)
Perform steps in Rails Upgrade Guide [1.3.1](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#moving-between-versions) through [1.5](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#configure-framework-defaults)
Address broken tests (if any)
Fix breaking changes introduced by Rails or updated dependencies

### Testing Plan
- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -7,16 +8,105 @@
 
 module CaseflowEfolder
   class Application < Rails::Application
-
     # Initialize configuration defaults for originally generated Rails version.
-    # config.load_defaults 5.1
-    
+    config.load_defaults 6.0
+    config.autoloader = :classic
+    config.active_record.cache_versioning = false
```
Comment (AKeyframe) [inline config/application.rb:14]: This addition resolves the RuntimeError:

  You're using a cache store that doesn't support native cache versioning.
  Your best option is to upgrade to a newer version of ActiveSupport::Cache::RedisStore
  that supports cache versioning (ActiveSupport::Cache::RedisStore.supports_cache_versioning? #=> true).

  Next best, switch to a different cache store that does support cache versioning:
  https://guides.rubyonrails.org/caching_with_rails.html#cache-stores.

  To keep using the current cache store, you can turn off cache versioning entirely:

      config.active_record.cache_versioning = false

File: config/environments/development.rb [hunk 2]
```diff
@@ -16,16 +16,16 @@
...
     config.action_controller.perform_caching = true
+    config.action_controller.enable_fragment_cache_logging = true
     config.public_file_server.headers = {
       'Cache-Control' => "public, max-age=#{2.days.to_i}"
     }
-
   else
     config.action_controller.perform_caching = false
 
   end
```
Comment (jcroteau) [inline config/environments/development.rb:26]: 💡 I wonder if we shouldn't add back in the two generated `config.cache_store` lines, but commented out, with a helpful comment referencing the commit where we removed them for historical context. That way, the next time we (or someone else) does an upgrade, we don't have to dig through the git history all over again to figure out why these lines were stripped out.

Maybe something like this:

```ruby
  if Rails.root.join('tmp', 'caching-dev.txt').exist?
    config.action_controller.perform_caching = true

    # Fall back to `application.rb` setting
    #   For historical context, see related commit d38aed96:
    #   https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/1030/commits/3d8e1061dc8b9e7dca8cb658f3af3a301a070df0
    # config.cache_store = :memory_store
    config.action_controller.enable_fragment_cache_logging = true
    config.public_file_server.headers = {
      'Cache-Control' => "public, max-age=#{2.days.to_i}"
    }
  else
    config.action_controller.perform_caching = false

    # Fall back to `application.rb` setting
    #   For historical context, see related commit d38aed96:
    #   https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/1030/commits/3d8e1061dc8b9e7dca8cb658f3af3a301a070df0
    # config.cache_store = :null_store
  end
```

We probably ought to have adopted this practice from the beginning, whenever we preserve some deviation from a default/generated setting. Something to consider as we go forward with future upgrades...

File: config/environments/test.rb [hunk 3]
```diff
@@ -41,15 +43,15 @@
...
 
 #=========================================================================================
 # eFolder - Custom Config Settings
 # Keep all efolder specific config settings below for clean diff's when upgrading rails
 #=========================================================================================
-  config.s3_enabled = false
+config.s3_enabled = false
 
-  config.api_key = "token"
+config.api_key = "token"
```
Comment (jcroteau) [inline config/environments/test.rb:56]: 🎨 I think the indentation was accidentally removed here as well.

**Review by jcroteau — COMMENTED** [review]
Really great work on this, and done in record time! I left a few comments. Aside from a few very minor changes, I think this is basically ready go, as soon as we ship the `5.2.8.1` upgrade.

After your changes, I will approve. And I think it'd be good to get another set of eyes on this as well.

(**Note:** I had to change the base branch to `AlecK/APPEALS-26726` for review purposes, to isolate your changes from those in that branch, but when we merge that PR, I believe the base branch on this PR should automatically resolve back to `master`)

File: config/environments/development.rb [hunk 4]
```diff
@@ -50,28 +50,28 @@
...
 
-  # Raises error for missing translations
-  # config.action_view.raise_on_missing_translations = true
+# Raises error for missing translations
+# config.action_view.raise_on_missing_translations = true
 
-  config.sqs_create_queues = true
-  config.sqs_endpoint = 'http://localhost:4566'
+config.sqs_create_queues = true
+config.sqs_endpoint = 'http://localhost:4566'
```
[the remark this answers is not in the document]
Comment (AKeyframe) [reply config/environments/development.rb:76]: Whoops, unsure how that happened. Fixed.

File: app/services/image_converter_service.rb [hunk 5]
```diff
@@ -10,10 +10,10 @@ def process
     return image if self.class.converted_mime_type(record.mime_type) == record.mime_type
 
     converted_image = convert
-    record.update_attributes!(conversion_status: :conversion_success)
+    record.update!(conversion_status: :conversion_success)
     converted_image
   rescue ImageConverterError
-    record.update_attributes!(conversion_status: :conversion_failed)
+    record.update!(conversion_status: :conversion_failed)
```
[the remark this answers is not in the document]
Comment (AKeyframe) [reply app/services/image_converter_service.rb:16]: Noted, I'll try and keep this in mind going forward with future updates.

File: config/environments/development.rb (hunk 2 above)
Comment (AKeyframe) [reply config/environments/development.rb:26]: Assuming it sticks around in 6.1 I'll be sure to add this when updating it next.

**Comment by jcroteau** [comment]
Superseded by https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/1478

**Pull request closed without merging**